serial.ts

```diff
@@ -367,9 +397,9 @@ class SerialPort {
    * @return {object} The interface Object created from the WebUSB API that is
    * has the specified classNumber
    */
-  getInterfaceWithClass_(device, classNumber) {
+  private getInterfaceWithClass(device: USBDevice, classNumber: number) {
     let interfaceWithClass;
-    device.configuration.interfaces.forEach((deviceInterface) => {
+    device.configuration!.interfaces.forEach((deviceInterface) => {
```

Comment:
What is the purpose of this "!"?

Reply:
It's an assertion that the field is non-null, which is guaranteed to be true if we've called `selectConfiguration()` however I notice that we call this method before `open()` is called. I've moved these calls into `open()` after the configuration is selected.